### PR TITLE
change workflow documentation

### PR DESCRIPTION
**Pull request by rrsettgast**
changes the documentation for the git workflow to go with a simplified gitflow model.

### CODE REVIEW

File: src/docs/sphinx/developerGuide/Contributing/GitWorkflow.rst [hunk 1]
```diff
@@ -124,40 +128,31 @@ should be created to perform the review and merging process.
...
-         \            / \
-          \          /   \
-          BA--------BB    EA     (release/1.2.x)
-                        (1.2.0)
+   A----B----C----D----E----F-----G               (release)
+        ^                         ^
+        |                         |
+        |                         |
+   A----B----C----D----E----F-----G------------   (develop)
+         \            / \        / (tag v1.2.0)
```
Comment (corbett5) [inline src/docs/sphinx/developerGuide/Contributing/GitWorkflow.rst:152]: The tags go on develop? I'm not sure it matters but I think it's easiest with a GitHub release to tag the actual release branch (release in this case).

**Review by corbett5 — APPROVED** [review]

File: src/docs/sphinx/developerGuide/Contributing/GitWorkflow.rst [hunk 2]
```diff
@@ -167,69 +162,30 @@ The main difference with a ``bugfix`` branch is that the primary target branch i
...
-
-`Documentation for git cherry-pick <https://git-scm.com/docs/git-cherry-pick>`_
+                        (tag v1.2.1)  (tag v1.3.0)
+   A----B'---...-----G---------H'-----------J'     (release)
+        ^            ^\       / \           ^
+        |            | \     /   \          |
+        |            |  GA--GB    \         |     (hotfix)
+        |            |             \        |
+   A----B----...-----G--------------H---I---J     (develop)
+                (tag v1.2.0)
```
Comment (TotoGaz) [inline src/docs/sphinx/developerGuide/Contributing/GitWorkflow.rst:183]: Should not the `hotfix` branch go to both `release` and `develop` (starting from `GB`), while there should be no direct link from `release` to `develop` (`H'` to `H`)

**Comment by CusiniM** [comment]
Are we gonna run nightly tests on the release branch as well?

**Comment by corbett5** [comment]
> 
> 
> Are we gonna run nightly tests on the release branch as well?

No, we shouldn't be changing it very often.

File: src/docs/sphinx/developerGuide/Contributing/GitWorkflow.rst (hunk 2 above)
Comment (rrsettgast) [reply src/docs/sphinx/developerGuide/Contributing/GitWorkflow.rst]: I have modified the diagram to show a proper merge-back.

File: src/docs/sphinx/developerGuide/Contributing/GitWorkflow.rst (hunk 1 above)
Comment (rrsettgast) [reply src/docs/sphinx/developerGuide/Contributing/GitWorkflow.rst]: Tags are related to commits, but we should tag commits from release branch even if they exist on develop.

**Pull request merged**